verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -0,0 +1,177 @@
+// Copyright 2021 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "verilog/tools/ls/symbol-table-handler.h"
+
+#include <filesystem>
+
+#include "common/strings/line_column_map.h"
+
+namespace verilog {
+
+static const std::string fileschemeprefix = "file://";
+
+bool LSPUriToPath(absl::string_view uri, std::string *path) {
```

Comment:
Given that we just return a substr() of a string_view, we can also make the output parameter a `string_view`.

(std::string will have to allocate memory, but string_view does not)